workspace/manifest-writer/src/index.ts

```diff
@@ -85,6 +96,57 @@ export async function addCatalogs (workspaceDir: string, newCatalogs: Catalogs):
     }
   }
 
+  if (shouldBeUpdated) {
+    await writeManifestFile(dir, manifest)
```

Comment:
Why do you need writeManifestFile here? It will be done in updateWorkspaceManifest

Reply:
It looks like the original logic has been adjusted.